engine/src/main/java/com/datatorrent/stram/engine/StreamingContainer.java

```diff
@@ -769,7 +769,11 @@ public void processHeartbeatResponse(ContainerHeartbeatResponse rsp)
     }
 
     if (rsp.committedWindowId != lastCommittedWindowId) {
+
       lastCommittedWindowId = rsp.committedWindowId;
+
+      bufferServer.purge(lastCommittedWindowId);
```

Comment:
Subscriber request comes with committed window Id. Buffer Server sends the data from the next higher window.